File: 2017/05b/jump.py
#!/usr/bin/env python
import sys
with open(sys.argv[1]) as input:
    instructions = [int(x) for x in input.readlines()]
jumps = 0
position = 0
exit = len(instructions)
while (position < exit):
    instruction = instructions[position]
    if instruction >= 3:
        instructions[position] = instruction - 1
    else:
        instructions[position] = instruction + 1
    position = position + instruction
    jumps += 1
print jumps

File: 2017/05b/jump.pl
#!/usr/bin/env perl

use strict;
use warnings;

chomp(my @instructions = <>);
my $jumpcount = 0;
my $pos = 0;
my $exit = scalar(@instructions);
while ($pos < $exit) {
    $pos += my $offset = $instructions[my $oldpos = $pos];
    ($instructions[$oldpos] >= 3) ?
        $instructions[$oldpos]-- :
        $instructions[$oldpos]++;
    $jumpcount++;
}
print "$jumpcount\n";
